Userland/Libraries/LibWeb/HTML/Canvas/CanvasPath.cpp

```diff
@@ -204,4 +204,157 @@ void CanvasPath::rect(float x, float y, float width, float height)
+{
+    // 1. If any of x, y, w, or h are infinite or NaN, then return.
+    if (!isfinite(x) || !isfinite(y) || !isfinite(w) || !isfinite(h))
+        return {};
+
+    // 2. If radii is an unrestricted double or DOMPointInit, then set radii to « radii ».
+    if (radii->has<double>() || radii->has<Geometry::DOMPointInit>()) {
+        Vector<Variant<double, Geometry::DOMPointInit>> radii_list;
+        if (radii->has<double>())
+            radii_list.append(radii->get<double>());
+        if (radii->has<Geometry::DOMPointInit>())
+            radii_list.append(radii->get<Geometry::DOMPointInit>());
+        radii = radii_list;
+    }
+
+    // 3. If radii is not a list of size one, two, three, or four, then throw a RangeError.
+    if (radii->get<Vector<Variant<double, Geometry::DOMPointInit>>>().size() > 4)
+        return WebIDL::SimpleException { WebIDL::SimpleExceptionType::RangeError, "roundRect: Can have between 1 and 4 radii"sv };
+
+    // 4. Let normalizedRadii be an empty list.
+    Vector<Geometry::DOMPointInit> normalized_radii;
+
+    // 5. For each radius of radii:
+    for (auto const& radius : radii->get<Vector<Variant<double, Geometry::DOMPointInit>>>()) {
+        // 5.1. If radius is a DOMPointInit:
+        if (radius.has<Geometry::DOMPointInit>()) {
+            // 5.1.1. If radius["x"] or radius["y"] is infinite or NaN, then return.
+            if (!isfinite(radius.get<Geometry::DOMPointInit>().x) || !isfinite(radius.get<Geometry::DOMPointInit>().y))
+                return {};
+
+            // 5.1.2. If radius["x"] or radius["y"] is negative, then throw a RangeError.
+            if (radius.get<Geometry::DOMPointInit>().x < 0 || radius.get<Geometry::DOMPointInit>().y < 0)
+                return WebIDL::SimpleException { WebIDL::SimpleExceptionType::RangeError, "roundRect: Radius can not be negative"sv };
+
+            // 5.1.3. Otherwise, append radius to normalizedRadii.
+            normalized_radii.append(radius.get<Geometry::DOMPointInit>());
+        }
+
+        // 5.2. If radius is a unrestricted double:
+        if (radius.has<double>()) {
+            // 5.2.1. If radius is infinite or NaN, then return.
+            if (!isfinite(radius.get<double>()))
+                return {};
+
+            // 5.2.2. If radius is negative, then throw a RangeError.
+            if (radius.get<double>() < 0)
+                return WebIDL::SimpleException { WebIDL::SimpleExceptionType::RangeError, "roundRect: Radius can not be negative"sv };
+
+            // 5.2.3. Otherwise append «[ "x" → radius, "y" → radius ]» to normalizedRadii.
+            normalized_radii.append(Geometry::DOMPointInit { radius.get<double>(), radius.get<double>() });
+        }
+    }
+
+    // 6. Let upperLeft, upperRight, lowerRight, and lowerLeft be null.
+    Geometry::DOMPointInit upper_left;
+    Geometry::DOMPointInit upper_right;
+    Geometry::DOMPointInit lower_right;
+    Geometry::DOMPointInit lower_left;
```

Comment:
I'm not 100% sure if these are initialized, just to make sure:
```suggestion
    Geometry::DOMPointInit upper_left {};
    Geometry::DOMPointInit upper_right {};
    Geometry::DOMPointInit lower_right {};
    Geometry::DOMPointInit lower_left {};
```